definitions/checks/repositories/check_upstream_repository.rb

```diff
@@ -0,0 +1,58 @@
+class Checks::CheckUpstreamRepository < ForemanMaintain::Check
+  metadata do
+    label :check_upstream_repository
+    description 'Check if any upstream repositories are enabled on system'
+    tags :pre_upgrade
+    preparation_steps do
+      Procedures::Packages::Install.new(:packages => %w[yum-utils])
+    end
+    confine do
+      feature(:downstream)
+    end
+  end
+
+  def run
+    with_spinner('Checking for presence of upstream repositories') do
+      enabled_upstream_repos = run_compare
+      repos_to_disable = comma_separated_repositories(enabled_upstream_repos)
+      assert(enabled_upstream_repos.empty?,
+             "System has upstream #{repos_to_disable} repositories enabled",
+             :next_steps => Procedures::Repositories::Disable.new(:repos => repos_to_disable))
+    end
+  end
+
+  def comma_separated_repositories(repos)
+    repos.map! { |r| r.gsub(%r{Repo-id:|\/+\w*}, '') }.join(',')
```

Comment:
Since you're immediately using the result of the `map!` I'd recommend using `map` which does not "mutate" the original array